### PR TITLE
deps: integrate with latest openshift version

### PR DESCRIPTION
**Pull request by ironcladlou**
* Upgrade to latest openshift/api and libary-go
* Upgrade to kube 1.13
* Upgrade to latest compatible controller-runtime

Fixes openapi spec generation, which as of the previous version was not picking
up field descriptions.

### CODE REVIEW

File: Gopkg.toml [hunk 1]
```diff
@@ -1,10 +1,6 @@
 # Force dep to vendor the code generators, which aren't imported just used at dev time.
 # Picking a subpackage with Go code won't be necessary once https://github.com/golang/dep/pull/1545 is merged.
 required = [
-  "k8s.io/code-generator/cmd/defaulter-gen",
```
Comment (danehans) [inline Gopkg.toml]: Are these code generators being removed b/c https://github.com/golang/dep/pull/1545 merged?

File: Gopkg.toml [hunk 2]
```diff
@@ -38,24 +34,34 @@ required = [
 
 [[constraint]]
   name = "github.com/openshift/library-go"
-  revision = "fc4b2d5cbcf0ab855422e340c78fe7a53a38bee3"
+  revision = "9067dee609b10991fa6c26b5d51fe095b2331027"
```
Comment (danehans) [inline Gopkg.toml]: Is there a reason for using this rev instead of latest (`dab26bb3a8dc7fccde7227194af755bbff30ce5d`)?

**Comment by ironcladlou** [comment]
https://github.com/kubernetes-sigs/controller-runtime/pull/383

**Comment by ironcladlou** [comment]
Need a resolution to https://github.com/kubernetes-sigs/controller-runtime/pull/383.

/hold

File: Gopkg.toml (hunk 2 above)
Comment (ironcladlou) [reply Gopkg.toml]: Typo on my part, good catch!

File: Gopkg.toml (hunk 1 above)
Comment (ironcladlou) [reply Gopkg.toml]: Latest commit cleans up the entire file including the `required` stanza. We still need some stuff in `required` for non-imported code generation, but the subpackage note isn't relevant to us.

**Comment by ironcladlou** [comment]
@openshift/sig-network-edge 

I don't want to wait for this. I'd like to carry the patched `Source` in our tree (e.g. in pkg/util) and duplicate the dependent controller-runtime packages to make it work. When/if upstream merges, we can remove our copy.

If there are no objections, I'll do this tomorrow (April 5).

**Comment by ironcladlou** [comment]
I figured, why wait. I went ahead and pushed the local Source implementation. If upstream patches it'll be trivial to delete our local version.

This is ready for review.

/hold cancel

**Comment by ironcladlou** [comment]
https://github.com/kubernetes-sigs/controller-runtime/pull/383 merged last night so I fixed up the PR to use it!

**Comment by Miciah** [comment]
Typo in the commit message: "libary-go".

**Comment by ironcladlou** [comment]
>Typo in the commit message: "libary-go".

Fixed

File: pkg/util/source/doc.go [hunk 3]
```diff
@@ -0,0 +1,7 @@
+// TODO: this package locally re-implements the following upstream patch:
+//
+// https://github.com/kubernetes-sigs/controller-runtime/pull/383
+//
+// If upstream accepts the patch, this package can be deleted and replaced by
+// upstream code.
+package source
```
Comment (Miciah) [inline pkg/util/source/doc.go]: Looks like your forgot to remove this file.

Comment (ironcladlou) [reply pkg/util/source/doc.go]: I must have failed to isolate those changed in the deleted commit. Thanks!

**Pull request merged**